great_expectations/validator/metrics_calculator.py

```diff
@@ -1,12 +1,16 @@
+from __future__ import annotations
+
 import logging
-from typing import Any, Dict, List, Optional, Set, Tuple, Union
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Set, Tuple, Union
 
-from great_expectations.execution_engine import ExecutionEngine  # noqa: TCH001
 from great_expectations.validator.computed_metric import MetricValue  # noqa: TCH001
 from great_expectations.validator.exception_info import ExceptionInfo  # noqa: TCH001
 from great_expectations.validator.metric_configuration import MetricConfiguration
 from great_expectations.validator.validation_graph import ValidationGraph
 
+if TYPE_CHECKING:
+    from great_expectations.execution_engine import ExecutionEngine
```

Comment:
Let's move the other `noqa` lines here please

Reply:
@cdkini Per our chat -- usually RUFF tells us to do this; right now, it is not -- most likely because these types are needed for more than type hints (e.g., to instantiate a class).  Thanks.